azure/Guidelines.md

```diff
@@ -846,110 +847,259 @@ https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#perfo
 
-<a href="#lro-no-patch-lro" name="lro-no-patch-lro">:no_entry:</a> **DO NOT** implement PATCH as an LRO.  If LRO update is required it must be implemented with POST.
+<a href="#lro-response-time" name="lro-response-time">:white_check_mark:</a> **DO** implement an operation as an LRO if the 99th percentile response time is greater than 1 second and when the client should poll the operation before making more progress.
 
-In rare instances where an operation may take a _very long_ time to complete, e.g. longer than 15 minutes,
-it may be better to expose this as a first class resource of the API rather than as an operation on another resource.
+<a href="#lro-no-patch-lro" name="lro-no-patch-lro">:no_entry:</a> **DO NOT** implement PATCH as an LRO. If LRO update semantics are required, implement it using the [LRO POST action pattern](#lro-existing-resource) .
 
-There are two basic patterns for long-running operations in Azure. The first pattern is used for a POST and DELETE
-operations that initiate the LRO. These return a `202 Accepted` response with a JSON status monitor in the response body.
-The second pattern applies only in the case of a PUT operation to create a resource that also involves additional long-running processing.
-For guidance on when to use a specific pattern, please refer to [Considerations for Service Design, Long Running Operations](./ConsiderationsForServiceDesign.md#long-running-operations).
-These are described in the following two sections.
+#### Patterns to Initiate a Long-Running Operation
 
-#### POST or DELETE LRO pattern
+<a href="#lro-valid-inputs-synchronously" name="lro-valid-inputs-synchronously">:white_check_mark:</a> **DO** perform as much validation as practical when initiating an LRO operation to alert clients of errors early.
 
-A POST or DELETE long-running operation accepts a request from the client to initiate the operation processing and returns
-a [status monitor](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.3) that reports the operation's progress.
+<a href="#lro-returns-operation-location" name="lro-returns-operation-location">:white_check_mark:</a> **DO** include an `operation-location` response header with the absolute URL of the status monitor for the operation.
 
-<a href="#lro-no-post-create" name="lro-no-post-create">:no_entry:</a> **DO NOT** use a long-running POST to create a resource -- use PUT as described below.
+<a href="#lro-operation-location-includes-api-version" name="lro-operation-location-includes-api-version">:ballot_box_with_check:</a> **YOU SHOULD** include the `api-version` query parameter in the `operation-location` response header with the same version passed on the initial request but expect a client to change the `api-version` value to whatever a new/different client desires it to be.
```

Comment:
We are a bit inconsistent within the guide about the `api-version` in location header.
https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md?plain=1#L264-L265 (here we say "without an api-version query parameter")

Reply:
As a User/Client of the API, when/why would you want to change the `api-version` being used for the LRO? I don't see the benefit - surely the API is going to know which LRO API versions support/can properly describe each Resource Type? Whilst I can see the benefit of changing the API version for regular endpoints - I suspect this would cause more problems than it solves?

As such - I'd suggest that this becomes a `MUST` rather than a `SHOULD`:

```suggestion
<a href="#lro-operation-location-includes-api-version" name="lro-operation-location-includes-api-version">:ballot_box_with_check:</a> **YOU MUST** include the `api-version` query parameter in the `operation-location` response header with the same version passed on the initial request but expect a client to change the `api-version` value to whatever a new/different client desires it to be.
```

Reply:
I'm a distributed system, it is common for one service using one api-version to initiate an LRO and then to hand the link to another service that may be using a different api-version which expects the over-the-wire schemas to match what it expects. Therefore, the calling client of any single operation must always use the api-version of that client. 

This is usually only an issue for patterns that require 2 or more operations which includes LROs and paging over collections.

Reply:
Thanks for catching that @weidongxu-microsoft. Fixed now.